### ISSUE TITLE
"Shaped" dither is specific to 44.1 kHz

### ISSUE THREAD

**Issue by dlh-0492** [audacity/audacity#3025]
**Describe the bug**
I've been doing some research into dithering and noise shaping, and I was wondering what the "shaped dither" really was. I looked into Dither.cpp and found that it was actually referring to _noise shaping._

So, to try to understand how everything was working, I generated four sinewaves at 44.1 kHz, 16 bit resolution, and amplitude of 1.0. Then I reduced the bit depth down to 4 bits by running two Nyquist prompts: `(mult *track* (recip 4096))` followed by `(mult *track* 2048)`. On the four different tracks, the first one was done with dither set to "none" under "high-quality conversion" before the bit depth was reduced. On the subsequent tracks, I used rectangular, triangular, and shaped dither.

Then I ran "Plot Spectrum" to see what was going on, and when I got to the one where I applied Shaped dither, I noticed that there are two EQ cuts in the noise spectrum, at 4 kHz and 12.3 kHz. However, these coefficients seem to be hard-coded for 44.1 kHz as the location of the cuts seems to be directly proportional to the sample rate. For example, when I repeated the aformentioned procedure at 48 kHz, the EQ cuts (with Shaped dither) were at 4.4 kHz and 13.5 kHz.

Should the cuts remain at 4 kHz and 12.3 kHz (where they are supposed to be) regardless of the sample rate? If so, could the dithering program in Audacity be modified as to resample the filter coefficients (2.033, -2.165, 1.959, -1.590, 0.6149) to adapt this filter for different sample rates? If not, is this unnecessary or undesirable and why?

**Additional information (please complete the following information):**
 - OS: Ubuntu
 - Version: 22.04
 - Sound Device: ALSA

**Comment by SteveDaulton** [comment]
I did some testing of this several years ago when I noticed this. From what I remember of my results:

At 44100 Hz, Audacity's shaped dither works pretty well. There's a trade-off between quality and complexity (processing speed). There are better (less audible) shaped dithers, but mostly they come at the expense of more complex processing. Audacity's shaped dither is a lot better than rectangle or triangle, not quite as good (a little more audible) than the best, but it is fast.

At 48000 Hz, the "notches" are a little off the "ideal" frequencies according to commonly quoted "loudness curves", but note that equal loudness curves are experimentally derived from hearing tests. In reality, hearing varies from person to person. There may not actually be anyone that has hearing that _exactly_ matches the theoretical curves. The ~10% difference in notch frequencies is probably not very significant. The famous studies by Fletcher and Munson, Churcher and King, and Robinson and Dadson, are all slightly different. The Fletcher-Munson curve is pretty flat at  20 phon between 4 - 5 kHz.

At 96 kHz and above, Audacity's shaped dither is "the wrong shape", but the noise has been shifted further into the extreme high frequencies. Although it is "the wrong shape", it is actually **quieter** than the "proper" shape.

In this quick illustration I have overlaid the 96  kHz curve (dark color) over the  44.1 kHz curve, for the frequency range 0 to 20 kHz.

![44 1 and 96](https://user-images.githubusercontent.com/11681097/185389801-dac8fdaa-a82d-4330-a6f4-cb712ceb2395.png)
Notice that the 96 kHz curve is a little _louder_ at 4 kHz, but a lot  quieter across the rest of the hearing range.
